ocaml/xapi/helpers.ml

```diff
@@ -2063,6 +2078,35 @@ let external_certificate_thumbprint_of_master ?(hash_type = `Sha256) () =
           | (_, cert_record) :: _ ->
               Some cert_record.certificate_fingerprint
       )
+  | `Sha1 ->
+      Server_helpers.exec_with_new_task
```

Comment:
Presumably this will can be quite frequent, I would rather have it available on the database.

The current certificate field can be deprecated, and replace it with 2, one per each hash. Then they can be filled in when creating the certificate, and can be requested in analogous ways.